rocAL/include/augmentations/geometry_augmentations/node_crop_resize.h

```diff
@@ -25,26 +25,42 @@ THE SOFTWARE.
 #include "parameters/parameter_crop_factory.h"
 #include "parameters/parameter_factory.h"
 
+#define NUM_ATTEMPTS 100
+
 class CropResizeNode : public CropNode {
    public:
     CropResizeNode(const std::vector<Tensor *> &inputs, const std::vector<Tensor *> &outputs);
     CropResizeNode() = delete;
     void init(float area, float aspect_ratio, float x_center_drift, float y_center_drift);
+    void init(std::vector<float>& area_factor, std::vector<float>& aspect_ratio,
+              RocalResizeInterpolationType interpolation_type = RocalResizeInterpolationType::ROCAL_LINEAR_INTERPOLATION);
     void init(FloatParam *area, FloatParam *aspect_ratio, FloatParam *x_drift_factor, FloatParam *y_drift_factor);
     void init(unsigned int crop_h, unsigned int crop_w, float x_drift, float y_drift,
               RocalResizeInterpolationType interpolation_type = RocalResizeInterpolationType::ROCAL_LINEAR_INTERPOLATION);
     unsigned int get_dst_width() { return _outputs[0]->info().max_shape()[0]; }
     unsigned int get_dst_height() { return _outputs[0]->info().max_shape()[1]; }
-    std::shared_ptr<CropParam> get_crop_param() { return _is_random_crop ? std::static_pointer_cast<CropParam>(_crop_param) : std::static_pointer_cast<CropParam>(_crop_fixed_param); }
+    std::shared_ptr<CropParam> get_crop_param() { 
+        if(_is_random_crop) {
+            if (_is_random_decode_crop) {
+                return std::static_pointer_cast<CropParam>(_crop_dec_param);
+            } else {
+                return std::static_pointer_cast<CropParam>(_crop_param);
+            }
+        } else {
+            return std::static_pointer_cast<CropParam>(_crop_fixed_param);
+        }
+        }
 
    protected:
     void create_node() override;
     void update_node() override;
 
    private:
     std::shared_ptr<RocalRandomCropParam> _crop_param;  // For random crop generation
+    std::shared_ptr<RocalRandomCropDecParam> _crop_dec_param;  // For random decode crop generation
```

Comment:
Instead of having multiple shared pointers, can we have one Base class pointer and create derived class instances and use accordingly

Most of the members called for these 3 params are similar
Please check the feasibility